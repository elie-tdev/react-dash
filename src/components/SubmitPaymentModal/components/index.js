export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
