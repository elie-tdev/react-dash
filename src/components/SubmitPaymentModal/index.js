export { default as SubmitPaymentModal } from './SubmitPaymentModal';
