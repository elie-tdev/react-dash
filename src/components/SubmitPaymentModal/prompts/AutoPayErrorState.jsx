import React from 'react';
import { useTranslation } from 'react-i18next';
import cn from 'classnames';
import { Box, DialogContent, Grid, Paper, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import { Alert } from '@material-ui/lab';
import WarningIcon from '@material-ui/icons/Warning';
import { useHistory } from 'react-router-dom';

import { PrimaryButton } from '@/components';

const useStyles = makeStyles(theme => ({
  confirmTitleContainer: {
    marginTop: '16px',
    marginBottom: '16px',
    textAlign: 'center',
  },
  confirmTitle: {
    display: 'inline',
    fontWeight: 500,
    fontSize: '20px',
    letterSpacing: '0.15px',
    lineHeight: '32px',
  },
  confirmSubTitle: {
    margin: '20px auto 30px',
    textAlign: 'center',
    fontSize: '14px',
    lineHeight: '20px',
    display: 'block',
    alignItems: 'center',
    letterSpacing: '0.1px',
  },
  confirmPanelBackdrop: {
    zIndex: 2000,
  },
  confirmPanelRoot: {
    borderRadius: '12px',
    backgroundColor: theme.palette.background.paper,
  },
  additionalAmountInput: {
    height: '92px',
    borderRadius: '8px',
  },
  additionalAmountInputLabel: {
    top: '20px',
  },
  button: {
    margin: '0px auto 40vh auto',
    height: '40px',
    lineHeight: '20px',
    fontSize: '14px',
    maxWidth: '120px',
    display: 'block',
    color: theme.palette.common.white,
    backgroundColor: theme.palette.primary.main,
  },
  chip: {
    padding: '0px 4px',
    border: '2px solid rgba(50, 70, 100, 0.1)',
    boxSizing: 'border-box',
    borderRadius: '12px',
    fontSize: '12px',
    height: '20px',
    fontWeight: '700',
    lineHeight: '17px',
  },
  autoPayOn: {
    borderColor: theme.palette.primary.main,
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.primary.contrastText,
  },
  paymentChips: {
    margin: '10px 0 0 0',
  },
  icon: {
    borderRadius: '50%',
    width: '50px',
  },
}));

const AutoPayErrorState = ({ loanNumber }) => {
  const classes = useStyles();
  const { t } = useTranslation();
  const history = useHistory();

  return (
    <Paper
      elevation={1}
      className={classes.confirmPanelRoot}
      PaperProps={{
        ['data-testid']: 'SubmitPaymentAutoPayErrorState',
      }}
    >
      <DialogContent>
        <Grid container justify={'space-evenly'} className={classes.paymentChips}>
          <Grid item>
            <Alert
              color={'error'}
              icon={<WarningIcon fontSize="inherit" />}
              className={classes.icon}
            ></Alert>
          </Grid>
        </Grid>
        <Box className={classes.confirmTitleContainer}>
          <Typography className={classes.confirmTitle} color={'textPrimary'} variant={'body2'}>
            {t('content.Your AutoPay details need to be reviewed') + '.'}
          </Typography>
        </Box>
        <Box>
          <Typography className={classes.confirmSubTitle} color={'textSecondary'} variant={'body2'}>
            {t('content.Please contact Customer Service for Support')}.
          </Typography>
        </Box>
      </DialogContent>
      <PrimaryButton
        className={cn(classes.button)}
        onClick={() => {
          history.push(`/dashboard/loans/${loanNumber}/helpCenter`);
        }}
      >
        {t('content.Contact Us')}
      </PrimaryButton>
    </Paper>
  );
};

export default AutoPayErrorState;
