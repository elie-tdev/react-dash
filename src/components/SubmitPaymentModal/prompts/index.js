export { default as SuccessPrompt } from './SuccessPrompt';
export { default as AdditionalAmountPrompt } from './AdditionalAmountPrompt';
export { default as CurrentlyEnrolledAutoPay } from './CurrentlyEnrolledAutoPay';
export { default as CurrentlyPastDue } from './CurrentlyPastDue';
export { default as PreExistingPaymentScheduled } from './PreExistingPaymentScheduled';
export { default as PayFromAccountPrompt } from './PayFromAccountPrompt';
export { default as SelectPaymentDatePrompt } from './SelectPaymentDatePrompt';
export { default as ConfirmPaymentDetails } from './ConfirmPaymentDetails';
export { default as LoadingDataError } from './LoadingDataError';
export { default as AutoPayErrorState } from './AutoPayErrorState';
export { default as CancelAutoPayPrompt } from './CancelAutoPayPrompt';
export { default as PaymentDisabledPrompt } from './PaymentDisabledPrompt';
