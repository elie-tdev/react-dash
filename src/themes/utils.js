export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
