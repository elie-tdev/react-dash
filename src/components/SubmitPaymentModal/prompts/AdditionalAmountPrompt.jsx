import React from 'react';
import { useTranslation } from 'react-i18next';
import { Backdrop, Box, DialogContent, Slide, Typography } from '@material-ui/core';
import Drawer from '@material-ui/core/Drawer';
import { makeStyles } from '@material-ui/styles';

import { prettyCurrency } from '@/utils/prettyHelpers';
import { PrimaryButton, SecondaryButton } from '@/components/Button';
import useGlobalStyles from '@/themes/useGlobalStyles';

const useStyles = makeStyles(theme => ({
  confirmTitleContainer: {
    marginTop: '16px',
    marginBottom: '16px',
    textAlign: 'left',
  },
  confirmTitle: {
    display: 'inline',
    fontWeight: 500,
    fontSize: '16px',
    letterSpacing: '0.1px',
    lineHeight: '24px',
  },
  confirmSubTitle: {
    margin: '20px auto 30px',
    textAlign: 'left',
    fontSize: '14px',
    lineHeight: '16px',
    display: 'block',
    alignItems: 'center',
    letterSpacing: '0.1px',
  },
  confirmSubTitle2: {
    margin: '15px auto 40px',
    textAlign: 'center',
    fontSize: '14px',
    lineHeight: '20px',
    display: 'block',
    alignItems: 'center',
    letterSpacing: '0.1px',
  },
  confirmPanelBackdrop: {
    zIndex: 2000,
  },
  confirmPanelRoot: {
    zIndex: 3000,
    position: 'relative',
  },
  button: {
    margin: '0px auto 40px auto',
    maxWidth: '400px',
    display: 'block',
  },
}));

const AdditionalAmountPrompt = props => {
  const classes = useStyles();
  const { t } = useTranslation();
  const globalClasses = useGlobalStyles();
  const { amount, totalDueAmount, open, handleAmountChosen, onClose } = props;

  return (
    <Backdrop open={open} className={classes.confirmPanelBackdrop}>
      <Slide direction="up" in={open} mountOnEnter unmountOnExit>
        <Drawer
          id={'confirmPanel'}
          variant={'permanent'}
          anchor={'bottom'}
          open={open}
          className={classes.confirmPanelRoot}
          PaperProps={{
            className: globalClasses.modalPaper,
            ['data-testid']: 'SubmitPaymentAdditionalAmount',
          }}
        >
          <DialogContent>
            <Box className={classes.confirmTitleContainer}>
              <Typography className={classes.confirmTitle} color={'textPrimary'} variant={'body2'}>
                {t('content.You will still owe') +
                  ' ' +
                  prettyCurrency(totalDueAmount - amount) +
                  ' ' +
                  t('content.after this payment')}
                .{' ' + t('Do you want to pay more to get back on track')}?
              </Typography>
            </Box>
            <Box className={classes.confirmTitleContainer}>
              <Typography color={'textSecondary'} variant={'body2'}>
                {t('Do you want to pay more to reduce your amount due?')}
              </Typography>
            </Box>
            <Box>
              <PrimaryButton
                className={classes.button}
                fullWidth={true}
                size={'medium'}
                onClick={() => handleAmountChosen(amount, false, true)}
              >
                {t('content.No, I will pay') + ' ' + prettyCurrency(amount) + '.'}
              </PrimaryButton>
              <SecondaryButton
                className={classes.button}
                fullWidth={true}
                size={'medium'}
                onClick={onClose}
              >
                {t('content.Yes, go back')}
              </SecondaryButton>
            </Box>
          </DialogContent>
        </Drawer>
      </Slide>
    </Backdrop>
  );
};

export default AdditionalAmountPrompt;
