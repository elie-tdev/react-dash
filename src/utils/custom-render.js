import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
