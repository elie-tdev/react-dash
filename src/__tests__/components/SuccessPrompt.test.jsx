import React from 'react';

import { SuccessPrompt } from '@/components/SubmitPaymentModal/prompts';
import { fireEvent, render } from '@/utils/custom-render';

describe('Payment Success Prompt', () => {
  test('renders Payment Success Prompt', async () => {
    const { container, findByText } = render(
      <SuccessPrompt
        open={true}
        autoPay={true}
        onSuccess={() => {}}
        onClose={() => {}}
        resetFlow={() => {}}
        paymentAmount={500}
        remainingBalance={9393.54}
        nextPayment={'Jan 17, 2021'}
        paidVia={{ type: 'Checking', number: '12340378' }}
        receiptEmail={'johndoe@gmail.com'}
      />,
    );
    fireEvent.click(await findByText('Back to Loan Overview'));
    expect(container).toMatchSnapshot();
  });
});
