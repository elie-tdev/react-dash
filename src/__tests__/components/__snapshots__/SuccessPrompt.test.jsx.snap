// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Success Prompt renders Payment Success Prompt 1`] = `
<div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-3"
    style="opacity: 1; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; webkit-transform: none; transform: none;"
  >
    <div
      class="MuiDrawer-root makeStyles-confirmPanelRoot-4 MuiDrawer-docked"
      id="confirmPanel"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-46 MuiDrawer-paperAnchorDockedBottom makeStyles-paperAnchorDockedBottom-5 MuiPaper-elevation0"
        data-testid="SubmitPaymentSuccess"
      >
        <div
          class="MuiBox-root MuiBox-root-47 makeStyles-header-6"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <div
            class="MuiBox-root MuiBox-root-48 makeStyles-checkIconContainer-8 makeStyles-checkIcon-9"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root makeStyles-confirmTitle-1 MuiTypography-body1"
          >
            Thank you for scheduling a payment
            !
          </p>
        </div>
        <div
          class="MuiPaper-root makeStyles-amountSplit-15 MuiPaper-elevation5 MuiPaper-rounded"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Payment Amount
              </p>
              <b>
                $500.00
              </b>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Next Payment
              </p>
              <b>
                Jan 17, 2021
              </b>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-49 makeStyles-root-50 MuiButton-contained makeStyles-button-13 MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
                data-testid="PrimaryButton"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Schedule Another Payment
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-51 makeStyles-root-52 MuiButton-contained makeStyles-button-13 MuiButton-disableElevation MuiButton-fullWidth"
                data-testid="SecondaryButton"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back to Loan Overview
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                *
                Your balance amount will be adjusted once we receive your payment
                .
                Payments take up to 3 business days to process
                .
                View Payments to check the status of your payment
                .
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-53 makeStyles-paymentDetails-16"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Payment Details
          </p>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-evenly"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              Payment Amount
              <br />
              Pay From
              <br />
              Pay On
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
              style="text-align: right;"
            >
              <span>
                $500.00
              </span>
              <br />
              <span />
               (
              )
              <br />
              <span>
                Jan 17, 2021
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
